src/sentry/middleware/ratelimit.py

```diff
@@ -39,11 +44,15 @@ def process_view(self, request, view_func, view_args, view_kwargs):
             request.will_be_rate_limited = True
             enforce_rate_limit = getattr(view_func.view_class, "enforce_rate_limit", False)
             if enforce_rate_limit:
+                error_message = getattr(
+                    view_func.view_class, "rate_limit_error_message", DEFAULT_ERROR_MESSAGE
+                )
                 return HttpResponse(
                     {
-                        "detail": f"You are attempting to use this endpoint too frequently. "
-                        f"Limit is {rate_limit_check_dict['limit']} requests in "
-                        f"{rate_limit_check_dict['window']} seconds"
+                        "detail": error_message.format(
```

Comment:
Using `format()`  is going to be really fragile. The error message will have to match the `limit` and `window` keys exactly and introducing any new keys will cause a `KeyError`.

Reply:
Just tested adding an unknown key. Good catch.